src/engine.rs

```diff
@@ -219,188 +195,58 @@ impl FileEngineInner {
         }
     }
 
-    // Rewrite inactive region's entries and key/value pairs,
-    // so the old files can be dropped ASAP.
-    #[allow(dead_code)]
-    fn rewrite_inactive(&self) -> bool {
+    fn rewrite_inactive_logs(&self, will_force_compact: &mut Vec<u64>) {
         let inactive_file_num = {
-            self.pipe_log
-                .files_before(self.cfg.cache_size_limit.0 as usize)
+            let total_size = self.pipe_log.total_size();
+            let garbage_ratio = self.cache_stats.garbage_ratio();
+            let rewrite_limit = (total_size as f64 * (1.0 - garbage_ratio)) as usize;
+            self.pipe_log.last_file_before(rewrite_limit)
```

Comment:
The write amplification will increase, the rewrite line and force compact should use different size watermarks.

Reply:
In the 3 TiKV deployment, let's assume most of these regions are active, if one TiKV shut down, all these raft logs will not trigger compact command, and the garbage_ratio will almost equal 1, then the total raft log size will increase and reach an extremely large total size. There will no compact command before the region's raft logs size reaches the 2/3 region size threshold.